dozer-tracing/src/helper.rs

```diff
@@ -0,0 +1,116 @@
+use dozer_types::chrono::{DateTime, Utc};
+use dozer_types::types::{Field, Record};
+use dozer_types::types::{FieldDefinition, FieldType, Schema, SchemaIdentifier, SourceDefinition};
+use opentelemetry::sdk::export::trace::SpanData;
+
+pub(crate) fn map_span_data(span_data: SpanData) -> (Record, Vec<Record>) {
+    let start_time: DateTime<Utc> = span_data.start_time.into();
+    let end_time: DateTime<Utc> = span_data.end_time.into();
+
+    let span_id = u64::from_be_bytes(span_data.span_context.span_id().to_bytes());
+    let span_record = Record {
+        schema_id: Some(SchemaIdentifier { id: 1, version: 1 }),
+        values: vec![
+            Field::UInt(span_id),
+            Field::Binary(span_data.span_context.trace_id().to_bytes().to_vec()),
+            Field::Text(span_data.name.to_string()),
+            Field::UInt(u64::from_be_bytes(span_data.parent_span_id.to_bytes())),
+            Field::Timestamp(start_time.into()),
+            Field::Timestamp(end_time.into()),
+        ],
+        version: None,
+    };
+
+    let mut events = vec![];
+    for evt in span_data.events {
+        let ts: DateTime<Utc> = evt.timestamp.into();
+        let record = Record {
+            schema_id: Some(SchemaIdentifier { id: 2, version: 1 }),
+            values: vec![
+                Field::UInt(span_id),
+                Field::Text(evt.name.to_string()),
+                Field::Timestamp(ts.into()),
+            ],
+            version: None,
+        };
+
+        events.push(record);
+    }
+    (span_record, events)
+}
+
+pub fn spans_schema() -> Schema {
+    let fields = vec![
+        FieldDefinition {
+            name: "id".to_string(),
+            typ: FieldType::UInt,
+            nullable: false,
+            source: SourceDefinition::Dynamic,
+        },
+        FieldDefinition {
+            name: "trace_id".to_string(),
+            typ: FieldType::Binary,
```

Comment:
We're going to change this to `UInt128` when we have it?

Reply:
yes